sdk/core/src/pipeline.rs

```diff
@@ -95,10 +125,10 @@ where
         self.http_client.as_ref()
     }
 
-    pub async fn send(
-        &self,
-        ctx: &mut PipelineContext<C>,
-        request: &mut Request,
+    pub async fn send<'a>(
+        &'a self,
+        ctx: &'a mut PipelineContext<'a, C>,
+        request: &'a mut Request,
```

Comment:
```suggestion
    pub async fn send(
        &self,
        ctx: &mut PipelineContext<'_, C>,
        request: &'a mut Request,
```

Reply:
Nice! I believe we also need to remove the `'a` from `&'a mut Request` though.

Reply:
Tried but does not work:

```
   Compiling azure_core v0.1.0 (/home/mindflavor/src/rust/azure-sdk-for-rust/sdk/core)
error[E0623]: lifetime mismatch
   --> sdk/core/src/pipeline.rs:134:19
    |
130 |         ctx: &mut PipelineContext<'_, C>,
    |                   ---------------------- this parameter and the returned future are declared with different lifetimes...
131 |         request: &mut Request,
132 |     ) -> Result<Response, Error> {
    |          -----------------------
    |          |
    |          this `async fn` implicitly returns an `impl Future<Output = Result<Response, Error>>`
133 |         self.pipeline[0]
134 |             .send(ctx, request, &self.pipeline[1..])
    |                   ^^^ ...but data from `self` is held across an await point here

For more information about this error, try `rustc --explain E0623`.
```

Reply:
Ah dang, not sure why that's the case. That's unfortunate :(